solution.cpp

```diff
@@ -1,138 +1,136 @@
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
-                      bool primes[],
-                      const string& infilename,
-                      const string& outfilename );
-
-// ----------------------------------------------------------------------------
+int find_max(const string& infilename);
 
-int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    // - Runtime sized arrays are a C99 feature, and a GCC and Clang extension.
-    //   Thus `bool primes[max+1];` is not standards compliant.  It should work
-    //   on Xcode (which uses Clang) and in Linux (where most people use either
-    //   GCC or Clang), but Visual Studio appears not to allow it.
-    bool * primes = (bool *) malloc( sizeof(bool) * (max+1) );
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
-    }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+int write_primes(const int size,
+	bool primes[],
+	const string& infilename,
+	const string& outfilename);
+
+int write_composites(const int size,
+	bool primes[],
+	const string& infilename,
+	const string& outfilename);
+
+int main()
+{
+	int max = find_max("input.txt");
+	if (max == -1) {
+		cout << "ERROR in 'find_max()'" << endl;
+		return 1;  
+	}
+
+	bool * primes = (bool *)malloc(sizeof(bool) * (max + 1));
+	sieve(max + 1, primes);
+
+	int ret;							// store return value...to check for error codes!
+	ret = write_primes(max + 1, primes, "input.txt", "primes.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_primes()`" << endl;
+		return 1;  
+	}
+
+	ret = write_composites(max + 1, primes, "input.txt", "composites.txt");
+	if (ret == -1) {
+		cout << "ERROR in `write_composites()`" << endl;
+		return 1;   
+	}
+
+	return 0;
 }
 
+
 // ----------------------------------------------------------------------------
 
 // TODO: write prototyped functions
 
+int find_max(const string& infilename)
+{
+	int max = 0,
+		x;
+	ifstream inputFile(infilename);
+	if (!inputFile.is_open())
+		return -1;									
+	while (inputFile >> x)
+	{
+		if (x > max)
+			max = x;
+	}
+	return max;
+}
+
+void sieve(const int size, bool primes[])
+{
+	primes[0] = false;        
+	primes[1] = false;							//0 and 1 is not a prime number, two is the first prime number
+	primes[2] = true;         
+	for (int i = 3; i < size; i++)              
+	{											//assign true or false to numbers in array
+		for (int j = 2; j < i; j++)
+		{
+			if (i%j == 0)
+				primes[i] = false;
+		}
+	}
```

Comment:
inefficient (and, this is not a correct implementation of the sieve of Eratosthenes), but it should work